makersacademy/example-story-website
# Data layer

**Previous architecture**
- The GraphQL `query` to find the `slugs` and `frontmatter` data was  in the `NavBar.` This data is passed to `NavBar`
- `TriangleLinks` component was a direct child of the the `NavBar`  
-  TriangleLinks` had another  GraphQL `query` to find `slugs` ( a repetition of the query in `NavBar`

**New architecture**
`NavigationContainer` which contains the: 
- GraphQL `query` to find the `slugs` and `frontmatter` data 
- Then passes this data to  its direct children`NavBar` and `Pagers` (`Pagers` is the new name for `TriangleLinks`)
- TriangleLinks has the children `PreviousPageLink` and `NextPageLink`
![image](https://user-images.githubusercontent.com/60652902/97432136-66850a00-1913-11eb-8081-406d81ebe3b5.png)

**Tests**
- Update props 
- Update what child components were expected
